### PR TITLE
Try connecting to Ubuntu keyserver on port 80 if default HKP port fails

### LINKED ISSUE #717: Python – Installation from source fails because GnuPG cannot import keys without user ID
**Update & tl;dr**
I figured our company firewall blocks connections to the default `hkp` port. The only `hkps` keyserver (to which connections are **not** blocked) specified in `install.sh` strips user IDs of some keys. GnuPG then refuses to import those keys. A solution could be to access the Ubuntu keyserver via `hkp` through port 80, which is not blocked. See my last comment: https://github.com/devcontainers/features/issues/717#issuecomment-1752455796

**Original post**
Installation of the `python` feature from source fails:

- At least one of the [specified key servers](https://github.com/devcontainers/features/blob/4fca96b5e8a4bfc93679098cb19d73c65ce571eb/src/python/install.sh#L31-L33) doesn't report the user ID for five of [the requested keys](https://github.com/devcontainers/features/blob/4fca96b5e8a4bfc93679098cb19d73c65ce571eb/src/python/install.sh#L30),
-  hence GnuPG refuses to import those keys:

```shell
 => => # (*) Downloading GPG key...                                            
 => => # gpg: key 3A5CA953F73C700D: new key but contains no user ID - skipped  
 => => # gpg: key 04C367C218ADD4FF: new key but contains no user ID - skipped  
 => => # gpg: key 0EDDC5F26A45C816: new key but contains no user ID - skipped  
 => => # gpg: key 6AF053F07D9DC8D2: new key but contains no user ID - skipped  
 => => # gpg: key C9BE28DEE6DF025C: new key but contains no user ID - skipped  
```

`devcontainer.json`:
```json
{
  "name": "dev",
  "image": "mcr.microsoft.com/devcontainers/base:debian-11",
  "features": {
    "ghcr.io/devcontainers/features/python:1": {
      "version": "3.11",
    },
  },
}
```

I can manually reproduce the problem on that container:
```shell
vscode ➜ ~ $ gpg --keyserver "hkps://keys.openpgp.org" --recv-keys 0x3A5CA953F73C700D
gpg: key 3A5CA953F73C700D: new key but contains no user ID - skipped
gpg: Total number processed: 1
gpg:           w/o user IDs: 1
```

GnuPG version:
```shell
vscode ➜ ~ $ gpg --version
gpg (GnuPG) 2.2.27
libgcrypt 1.8.8
Copyright (C) 2021 Free Software Foundation, Inc.
License GNU GPL-3.0-or-later <https://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Home: /home/vscode/.gnupg
Supported algorithms:
Pubkey: RSA, ELG, DSA, ECDH, ECDSA, EDDSA
Cipher: IDEA, 3DES, CAST5, BLOWFISH, AES, AES192, AES256, TWOFISH,
        CAMELLIA128, CAMELLIA192, CAMELLIA256
Hash: SHA1, RIPEMD160, SHA256, SHA384, SHA512, SHA224
Compression: Uncompressed, ZIP, ZLIB, BZIP2
```

Any help would be greatly appreciated.

### PR DESCRIPTION
**Pull request by hoechenberger**
Closes #719
Closes #717

cc @samruddhikhandale

### CODE REVIEW

**Comment by hoechenberger** [comment]
Oh noes, I'm realizing some autoformatting changes sneaked in there. Shall I remove them?

**Comment by momohnasirumuri** [comment]
Run

**Comment by hoechenberger** [comment]
This is ready for review.

File: src/git-lfs/install.sh [hunk 1]
```diff
@@ -15,6 +15,7 @@ GIT_LFS_ARCHIVE_ARCHITECTURES="amd64 arm64"
 GIT_LFS_ARCHIVE_VERSION_CODENAMES="stretch buster bullseye bionic focal jammy"
 GIT_LFS_CHECKSUM_GPG_KEYS="0x88ace9b29196305ba9947552f1ba225c0223b187 0x86cd3297749375bcf8206715f54fe648088335a9 0xaa3b3450295830d2de6db90caba67be5a5795889"
 GPG_KEY_SERVERS="keyserver hkp://keyserver.ubuntu.com
+keyserver hkp://keyserver.ubuntu.com:80
```
Comment (samruddhikhandale) [inline src/git-lfs/install.sh:18]: Can you bump minor versions for all the updated Features? Thanks!

**Review by samruddhikhandale — COMMENTED** [review]
@hoechenberger did this change work with your local config/machine? The tests are passing, so should be safe to merge it in. Also, I'll shortly test these changes on my local machine.

Left one minor comment.

**Comment by samruddhikhandale** [comment]
> @hoechenberger did this change work with your local config/machine? 

I saw your PR description update, looks like this PR changes are working for you?  🎉

**Comment by hoechenberger** [comment]
@samruddhikhandale I was struggling to get the tests to work on my corporate machine because I need to inject custom SSL certificates. I have written a custom feature to do that, but didn't have time to modify the tests to actually use this feature. So my honest answer is: no, I could not fully run the test suite yet. LMK if you would still like to go ahead and merge (after which it will be much easier for me to run the tests) and we'll have to revert if necessary; or otherwise I'll have to get the test suite to run locally, but it may take a couple of days (as I'm very busy today and probably also tomorrow)

**Comment by samruddhikhandale** [comment]
@hoechenberger I have a simpler way to test these changes, can you check if you can create a dev container with https://github.com/samruddhikhandale/test-gpg-keyserver/tree/main ? 

It adds the same dev configuration as mentioned in https://github.com/devcontainers/features/issues/717, the only difference is that it doesn't pull the python Feature from `ghcr`, instead it uses a local Feature (which has these ^ PR changes)

It would be great to know if this changes fixes your issue before we roll out this change. It works good with my testing, however, the gpg issue behaves differently on different machines (don't want to accidentally cause Feature failure blips - given that this PR updates seven Features)

**Comment by hoechenberger** [comment]
@samruddhikhandale Thank you! I will test this on my macOS machine and ask colleagues of mine to test on Linux and Windows too. I will get back to you once I have the test results.

File: src/git-lfs/install.sh (hunk 1 above)
Comment (hoechenberger) [reply src/git-lfs/install.sh:18]: Done!

**Comment by hoechenberger** [comment]
@samruddhikhandale Tested and working on two separate macOS computers. I couldn't test on other operating systems, unfortunately!

**Review by samruddhikhandale — APPROVED** [review]
Looks good to me~ Thank you!

**Pull request merged**